subscription/src/subscription_worker.py

```diff
@@ -39,18 +40,24 @@ def delete_messages(sqs_client, queue_url, messages):
 def process_messages(sns_client, topic, messages, access_control):
     """ Processes a list of messages that was received from a queue. Check to see if ACLs pass for the granule.
         If the checks pass then send the notification. """
-    for message in messages.get("Messages", []):
-
-        # Get the permission for the collection from access-control
-        # response = access_control.get_permissions(subscriber-id, collection-concept-id)
-        # Return is either None (Null or Nil) (if check on response is false) or
-        # {"C1200484253-CMR_ONLY":["read","update","delete","order"]}
-        #if response and if array contains read:
-        #    publish message.
-        #else: 
-        #    log subscriber-id no longer has read access to collection-concept-id
 
-        sns_client.publish_message(topic, message)
+    for message in messages.get("Messages", []):
+        try:
+            message_body = json.loads(message["Body"])
+            message_attributes = message_body["MessageAttributes"]
+            logger.debug(f"Subscription worker: Received message including attributes: {message_body}")
+
+            subscriber = message_attributes['subscriber']['Value']
+            collection_concept_id = message_attributes['collection-concept-id']['Value']
+
+            if(access_control.has_read_permission(subscriber, collection_concept_id)):
+                logger.debug(f"Subscription worker: {subscriber} has permission to receive granule notifications for {collection_concept_id}")
+                sns_client.publish_message(topic, message)
+            else:
```

Comment:
Looks like the message is just being ignored, do we need to remove the subscription so it doesn't take up future resources?

Reply:
Yes, the lack of permission is being logged, but that it is.  We could, but sometimes providers mess up their ACLs and fix them later.  I have the same question if the endpoint we are sending it to goes away, SQS is deleted or URL no longer works, do we remove the subscription?  May be after a length of time for the same error?  I can write a ticket for us to discuss this, but I think for this ticket, its too much work.

Reply:
I created issue https://bugs.earthdata.nasa.gov/browse/CMR-10415